src/geometry/rectangle.rs

```diff
@@ -0,0 +1,186 @@
+                             plane_constant: 0.,         //TODO fix this one
+                             gloabl_center: orig,
+                             to_global: to_global_transform,
+                             to_local: to_local_transform,
+                             measurement_to_state_vector: projection_mat};
+                             
+                dbg!{&rect};
+                Ok(rect)
+                },
+            None => return Err(MatrixError::NonInvertible)
+
+        }
+    }
+
+}
+impl Transform for Rectangle{
+    /*
+    /// Converts a point in the global reference frame to a point in the local reference frame of the sensor.
+    /// 
+    /// # Examples
+    /// ```
+    /// use nalgebra as na;
+    /// use na::Point3;
+    /// use nalgebra as na;
+    /// use kalman_rs::geometry::Rectangle;
+    /// 
+    /// let base_len = 3.0;
+    /// let height_len = 3.0;
+    /// let tfm_matrix= na::Matrix4::<f64>::identity(); //arbitrary transform matrix
+    /// let projection = na::Matrix5x2::<f64>::zeros();
+    /// Rectangle::new(base_len, height_len, tfm_matrix, projection).unwrap();
+    /// 
+    /// let global_point = rectangle_sensor.to_global(na::Point3::new(1.0, 2.0, 0.0));
+    /// ```*/
+    fn to_global(&self, input_point: P3)-> P3{
+        self.to_global * input_point
+    }
+    
+    /*
+    /// Converts a point in the local refernce frame of the sensor to the global reference frame.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// use nalgebra as na;
+    /// use na::Point3;
+    /// use kalman_rs::geometry::rectangle::Rectangle;
+    /// use kalman_rs::sensor_traits::Transform;
+    /// 
+    /// let base = 3.0;
+    /// let height = 3.0;
+    /// let tfm_matrix : na::Matrix4<f64>= na::Matrix4::new(1.0,0.0,0.0,0.0,  0.0,1.0,0.0,0.0,  0.0,0.0,1.0,0.0, 0.0,0.0,0.0,1.0);
+    /// let mut rectangle_sensor = Rectangle::new(base, height, tfm_matrix).unwrap();
+    /// 
+    /// let global_point = rectangle_sensor.to_local(na::Point3::new(6.0, 3.0, 5.0));
+    /// ```*/
+    fn to_local(&self, input_point: P3) -> P2{
+        let local = self.to_local * input_point;
+        return P2::new(local.x, local.y)
+    }
+
+    /*
+    /// Checks if a local point is contained within the bounds of a sensor.
+    /// 
+    /// # Examples
+    /// ```
+    /// use nalgebra as na;
+    /// use na::Point3;
+    /// use kalman_rs::sensor_traits::Transform;
+    /// use kalman_rs::geometry::rectangle::Rectangle;
+    /// 
+    /// let base = 3.0;
+    /// let height = 3.0;
+    /// let tfm_matrix : na::Matrix4<f64>= na::Matrix4::new(1.0,0.0,0.0,0.0,  0.0,1.0,0.0,0.0,  0.0,0.0,1.0,0.0, 0.0,0.0,0.0,1.0);
+    /// let mut rectangle_sensor = Rectangle::new(base, height, tfm_matrix).unwrap();
+    /// 
+    /// let is_point_on_sensor = rectangle_sensor.contains_from_local(&na::Point2::new(1.0, 6.0));
+    /// ```*/
+    fn contains_from_local(&self, input: &P2) -> bool {
```

Comment:
Maybe just call it `inside()`?

Reply:
changed:
`contains_from_local()` => `inside()`
`contains_from_global()` => `inside_global()`